x-pack/plugins/ml/server/saved_objects/repair.ts

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from '@hapi/boom';
+import { IScopedClusterClient } from 'kibana/server';
+import type { JobObject, JobSavedObjectService } from './service';
+import { JobType } from '../../common/types/saved_objects';
+import { checksFactory } from './checks';
+
+import { Datafeed } from '../../common/types/anomaly_detection_jobs';
+
+export function repairFactory(
+  client: IScopedClusterClient,
+  jobSavedObjectService: JobSavedObjectService
+) {
+  const { checkStatus } = checksFactory(client, jobSavedObjectService);
+
+  async function repairJobs(simulate: boolean = false) {
+    type Result = Record<string, { success: boolean; error?: any }>;
+    const results: {
+      savedObjectsCreated: Result;
+      savedObjectsDeleted: Result;
+      datafeedsAdded: Result;
+      datafeedsRemoved: Result;
+    } = {
+      savedObjectsCreated: {},
+      savedObjectsDeleted: {},
+      datafeedsAdded: {},
+      datafeedsRemoved: {},
+    };
+
+    const { body: datafeeds } = await client.asInternalUser.ml.getDatafeeds<{
+      datafeeds: Datafeed[];
+    }>();
+
+    const tasks: Array<() => Promise<void>> = [];
+
+    const status = await checkStatus();
+    for (const job of status.jobs['anomaly-detector']) {
```

Comment:
nit: as there is no `break` statements or `async` execution, what about using `.filter` and `.forEach`